api/room_resource/schema.py

```diff
@@ -164,16 +164,24 @@ class Arguments:
 
     @Auth.user_roles('Admin')
     def mutate(self, info, resource_id, **kwargs):
-        query_room_resource = Resource.get_query(info)
-        active_resources = query_room_resource.filter(
+        query_resources = Resource.get_query(info)
+        active_resources = query_resources.filter(
             ResourceModel.state == "active")
-        exact_room_resource = active_resources.filter(
+        exact_resource = active_resources.filter(
             ResourceModel.id == resource_id).first()
-        if not exact_room_resource:
+        if not exact_resource:
             raise GraphQLError("Resource not found")
-        update_entity_fields(exact_room_resource, state="archived", **kwargs)
-        exact_room_resource.save()
-        return DeleteResource(resource=exact_room_resource)
+        update_entity_fields(exact_resource, state="archived", **kwargs)
+        query_room_resource = RoomResource.get_query(info)
+        room_resources = query_room_resource.filter(
+            RoomResourceModel.resource_id == resource_id).all()
+        for room_resource in room_resources:
```

Comment:
Nice work @TeamoreA . Kindly shade more colour regarding this lines, you could make use of the background context section .

Reply:
@TeamoreA I also feel that the loop is a little vague. Something to do with the naming of `room_resource` and `room_resources`. I feel like they are little too similar to differentiate

Reply:
@TeamoreA Nice work on the implementation. I think you need to make some improvements on the PR description. It talks about `room(s) that were assigned to the resource...` but instead, we assign resources to rooms.  It also implies the rooms will be cleared whenever the resource is deleted, but I think you meant that the resource will be unassigned from the rooms whenever the resource is deleted.

Reply:
Otherwise nice implementation. It works as expected

Reply:
> Nice work @TeamoreA . Kindly shade some colour regarding this lines, you could make use of the background context section . It might take a minute for someone to get what the code is trying to achieve here.

This has been worked on @verenceLola

Reply:
> @TeamoreA Nice work on the implementation. I think you need to make some improvements on the PR description. It talks about `room(s) that were assigned to the resource...` but instead, we assign resources to rooms. It also implies the rooms will be cleared whenever the resource is deleted, but I think you meant that the resource will be unassigned from the rooms whenever the resource is deleted.

This has been worked on @rajeman